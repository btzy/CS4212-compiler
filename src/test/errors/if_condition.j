class Main {

Void main(){
    Int i; Int j; String cond;
    Int ir;
    String sr;
    if (i) {
        j = 1;
    }
    else {
        j = 2;
    }
    while (cond + "b") { cond = cond + "a"; }
}
}