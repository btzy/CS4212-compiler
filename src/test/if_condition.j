class Main {

Void main(){
    Int i; Int j; String k;
    Int ir;
    String sr;
    if (i) {
        j = 1;
    }
    else {
        j = 2;
    }
}
}