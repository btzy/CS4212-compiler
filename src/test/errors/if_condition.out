Error: Expected expression to have type "Bool", but got type "Int" instead ( line 7 col 9 - line 7 col 9 ) :
7 |     if (i) {
  |         ^
Note: Condition of if-statement must have type "Bool"
Error: Expected expression to have type "Bool", but got type "String" instead ( line 13 col 12 - line 13 col 21 ) :
13 |     while (cond + "b") { cond = cond + "a"; }
   |            ^^^^^^^^^^
Note: Condition of while-statement must have type "Bool"
Semantic checking aborted due to above errors.
